sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -196,4 +178,39 @@ export function record(
       return date;
     }
   };
+
+  const stringTest = testContext.currentTest!.fn!.toString();
+  const currentHash = MD5(stringTest);
+  const testAbsolutePath = testContext.currentTest!.file!;
+
+  if (
+    isSoftRecordMode() &&
+    !testHasChanged(testHierarchy, testTitle, testAbsolutePath, currentHash)
+  ) {
+    testContext.skip();
```

Comment:
Along with this, update the test title saying that we are skipping the test because the test hasn't changed.

`testContext.test!.title = testContext.test!.title + '# reason'`

Reply:
I pushed this. Let me know how it looks.